src/StudioComponents/Checkbox.lua

```diff
@@ -148,11 +148,8 @@ return function(props: CheckboxProperties): Frame
 							return if unwrap(isIndeterminate) then Color3.fromRGB(255, 255, 255) else indicatorColor
 						end),
 						ImageRectOffset = Computed(function()
-							local currentTheme = unwrap(themeProvider.Theme)
 							if unwrap(isIndeterminate) then
-								-- should probably check for the brightness of the background color and determine
-								-- which indeterminate color to use instead of checking for "Dark"
-								return if currentTheme=="Dark" then Vector2.new(13, 0) else Vector2.new(26, 0)
+								return themeProvider.IsDark:get() and Vector2.new(13, 0) or Vector2.new(26, 0)
```

Comment:
The themeProvider.IsDark state object should be defined at the top of the computed so it's captured as a dependency. This is because it's entirely possible for the isIndeterminate state object to be false.

Also, it's preferred to use unwrap() instead of :get() just in case use() eventually becomes a thing for computeds. Because if it does become a thing, unwrap would become use and Computed(function() would become Computed(function(use). You can't easily replace StateObject:get() to use(StateObject) to my knowledge with find and replace since a lot of state objects have different names.

Reply:
> The themeProvider.IsDark state object should be defined at the top of the computed so it's captured as a dependency. This is because it's entirely possible for the isIndeterminate state object to be false.

Part of what makes Fusion so cool is that you can have dynamic dependencies- if isIndeterminate is false, then we don't need to be depending on IsDark at the moment since it wouldn't change our UI anyway!

> Also, it's preferred to use unwrap() instead of :get() just in case use() eventually becomes a thing for computeds. Because if it does become a thing, unwrap would become use and Computed(function() would become Computed(function(use). You can't easily replace StateObject:get() to use(StateObject) to my knowledge with find and replace since a lot of state objects have different names.

Good point, will do in a moment. For what it worth though, you can use regex in VSCode's find & replace so it remains trivial to replace with a simple pattern.

Reply:
Huh, I was under the impression that dependencies were only captured the first time a computed ran.